src/nvim/window.c

```diff
@@ -2177,8 +2177,10 @@ winframe_remove (
         }
       }
     }
-    frame_new_height(frp2, frp2->fr_height + frp_close->fr_height,
-        frp2 == frp_close->fr_next ? TRUE : FALSE, FALSE);
+    frame_new_height(frp2,
+                     frp2->fr_height + frp_close->fr_height + (
+                         (!win->w_status_height) ? STATUS_HEIGHT : 0),
+                     frp2 == frp_close->fr_next ? true : false, false);
```

Comment:
`frp2 == frp_close->fr_next ? true : false` can be just `(frp2 == frp_close->fr_next)`.